sinabs/backend/dynapcnn/nir_graph_extractor.py

```diff
@@ -0,0 +1,325 @@
+                        new_edge = (edge[0], _trg)
+            elif isinstance(self.modules_map[_trg], default_ignored_nodes):
+                removed_nodes.append(_trg)
+                # all edges where node '_trg' is source change it to node '_src' as their source.
+                for edge in edges:
+                    if edge[0] == _trg:
+                        new_edge = (_src, edge[1])
+            else:
+                new_edge = (_src, _trg)
+            
+            if new_edge not in parsed_edges:
+                parsed_edges.append(new_edge)
+
+        removed_nodes = list(set(removed_nodes))
+
+        # remapping nodes indexes.
+        remapped_nodes = {}
+        for node_indx, __ in self.modules_map.items():
+            _ = [x for x in removed_nodes if node_indx > x]
+            remapped_nodes[node_indx] = node_indx - len(_)
+            
+        for x in removed_nodes:
+            del remapped_nodes[x]
+
+        # remapping nodes names in parsed edges.
+        remapped_edges = []
+        for edge in parsed_edges:
+            remapped_edges.append((remapped_nodes[edge[0]], remapped_nodes[edge[1]]))
+
+        return remapped_edges, remapped_nodes
+    
+    # TODO - it would be good if I/O shapes were returned by the NIR graph.
+    def get_node_io_shapes(self, node: int) -> Tuple[torch.Size, torch.Size]:
+        """ Returns the I/O tensors' shapes of `node`.
+
+        Returns
+        ----------
+        - input shape (torch.Size): shape of the input tensor to `node`.
+        - output shape (torch.Size): shape of the output tensor from `node`.
+        """
+        return self._nodes_io_shapes[node]['input'], self._nodes_io_shapes[node]['output']
+
+    ####################################################### Pivate Methods #######################################################
+
+    def _get_edges_from_nir(self, nir_graph: nirtorch.graph.Graph) -> Tuple[List[Tuple[int, int]], Dict[str, int], List[int]]:
+        """ Standardize the representation of `nirtorch.graph.Graph` into a list of edges (`Tuple[int, int]`) where
+        each node in `nir_graph` is represented by an interger (with the source node starting as `0`).
+
+        Parameters
+        ----------
+        - nir_graph (nirtorch.graph.Graph): a NIR graph representation of `spiking_model`.
+        
+        Returns
+        ----------
+        - edges_list (list): tuples describing the connections between layers in `spiking_model`.
+        - name_2_indx_map (dict): `key` is the original variable name for a layer in `spiking_model` and `value
+            is an integer representing the layer in a standard format.
+        - entry_nodes (list): IDs of nodes acting as entry points for the network (i.e., receiving external input).
+        """
+        edges_list = []
+        name_2_indx_map = {}
+        idx_counter = 0                                         # TODO maybe make sure the input node from nir always gets assined `0`.
```

Comment:
You could just use the length of `name_2_indx_map` instead of the `idx_counter`. This way it is always unique.

Reply:
@Willian-Girao Please review and resolve if appropriate.